<launch>
  <!-- door actions -->
  <node pkg="pr2_doors_actions" type="action_runner" name="door_actions">
   <!-- params for cartesian motion using ik -->
   <param type="double" name="pose_suggestion/shoulder_pan" value="1.05" />
   <param type="double" name="pose_suggestion/shoulder_lift" value="0.72" />
   <param type="double" name="pose_suggestion/upper_arm_roll" value="0.0" />
   <param type="double" name="pose_suggestion/elbow_flex" value="-1.78" />
   <param type="double" name="pose_suggestion/forearm_roll" value="-2.07" />
   <param type="double" name="pose_suggestion/wrist_flex" value="-1.57" />
   <param type="double" name="pose_suggestion/wrist_roll" value="-0.56" />
  </node>

  <!-- params for move base door costmap -->
  <group ns="door_actions/costmap_move_base_door" >
    <rosparam file="$(find pr2_doors_actions)/launch/footprint.yaml" command="load" />
    <!-- Start Costmap Parameters -->
    <param name="global_frame" value="odom_combined" />
    <param name="robot_base_frame" value="base_link" />
    <param name="update_frequency" value="5.0" />
    <param name="publish_frequency" value="1.0" />
    <param name="obstacle_range" value="1.0" />
    <param name="max_obstacle_height" value="2.0" />
    <param name="raytrace_range" value="1.5" />
    <param name="observation_sources" value="base_scan base_scan_marking" />

    <param name="base_scan/observation_persistence" value="0.0" />
    <param name="base_scan/expected_update_rate" value="0.2" />
    <param name="base_scan/data_type" value="LaserScan" />
    <param name="base_scan/clearing" value="true" />
    <param name="base_scan/marking" value="false" />
    <param name="base_scan_marking/observation_persistence" value="0.0" />
    <param name="base_scan_marking/expected_update_rate" value="0.2" />
    <param name="base_scan_marking/data_type" value="PointCloud" />
    <param name="base_scan_marking/clearing" value="false" />
    <param name="base_scan_marking/marking" value="true" />

    <param name="rolling_window" value="true" />

    <param name="static_map" value="false" />
    <!-- begin only used if static map is set to false -->
    <param name="width" value="5.0" />
    <param name="height" value="5.0" />
    <param name="origin_x" value="-2.5" />
    <param name="origin_y" value="-2.5" />
    <param name="resolution" value="0.01" />
    <!-- end only used if static map is set to false -->

    <param name="cost_scaling_factor" value="1.0" />
    <param name="lethal_cost_threshold" value="100" />
    <!-- End Costmap Parameters -->
    <remap from="obstacles" to="~obstacles" />
    <remap from="inflated_obstacles" to="~inflated_obstacles" />
  </group>

  <!-- arm ik action server -->
  <node pkg="pr2_arm_move_ik" type="arm_ik" name="r_arm_ik" output="screen">
    <param name="joint_trajectory_action" value="r_arm_controller/joint_trajectory_generator" />
    <param name="arm" value="r" />
    <param name="free_angle" value="2" />
    <param name="search_discretization" value="0.01" />
    <param name="ik_timeout" value="5.0" />
  </node>
  
</launch>
